src/System.Transactions/src/System/Transactions/Trace/TraceRecords.cs

```diff
@@ -0,0 +1,1306 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Xml;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace System.Transactions.Diagnostics
+{
+    internal enum EnlistmentType
+    {
+        Volatile = 0,
+        Durable = 1,
+        PromotableSinglePhase = 2
+    }
+
+    internal enum NotificationCall
+    {
+        // IEnlistmentNotification
+        Prepare = 0,
+        Commit = 1,
+        Rollback = 2,
+        InDoubt = 3,
+        // ISinglePhaseNotification
+        SinglePhaseCommit = 4,
+        // IPromotableSinglePhaseNotification
+        Promote = 5
+    }
+
+    internal enum EnlistmentCallback
+    {
+        Done = 0,
+        Prepared = 1,
+        ForceRollback = 2,
+        Committed = 3,
+        Aborted = 4,
+        InDoubt = 5
+    }
+
+    internal enum TransactionScopeResult
+    {
+        CreatedTransaction = 0,
+        UsingExistingCurrent = 1,
+        TransactionPassed = 2,
+        DependentTransactionPassed = 3,
+        NoTransaction = 4
+    }
+
+    /// <summary>
+    /// TraceHelper is an internal class that is used by TraceRecord classes to write
+    /// TransactionTraceIdentifiers and EnlistmentTraceIdentifiers to XmlWriters.
+    /// </summary>
+    internal static class TraceHelper
```

Comment:
In general, yes, there is, though I didn't do much of that with this PR.  In particular for this tracing stuff, I wasn't sure what the desired path forward would be, e.g. I thought you might be switching some or all of this to EventSource-based tracing, and figured effort to redo it could be done at the same time as changing the tracing mechanism.